src/clj/rems/event_notification.clj

```diff
@@ -0,0 +1,74 @@
+(ns rems.event-notification
+  (:require [clj-http.client :as http]
+            [clj-time.core :as time]
+            [clojure.test :refer :all]
+            [clojure.tools.logging :as log]
+            [mount.core :as mount]
+            [rems.api.schema :as schema]
+            [rems.config]
+            [rems.db.applications :as applications]
+            [rems.db.outbox :as outbox]
+            [rems.json :as json]
+            [rems.scheduler :as scheduler]
+            [rems.util :refer [getx]]))
+
+(def ^:private default-timeout 60)
+
+(defn- notify! [target body]
+  (log/info "Sending event notification to" (:url target))
+  (try
+    (let [timeout-ms (* 1000 (get target :timeout default-timeout))
+          response (http/put (getx target :url)
+                             {:body body
+                              :throw-exceptions false
+                              :content-type :json
+                              :headers (get target :headers)
+                              :socket-timeout timeout-ms
+                              :conn-timeout timeout-ms})
+          status (:status response)]
+      (when-not (= 200 status)
+        (log/error "Event notification response status" status)
+        (str "failed: " status)))
+    (catch Exception e
+      (log/error "Event notification failed" e)
+      "failed: exception")))
+
+(defn process-outbox! []
+  (doseq [entry (outbox/get-entries {:type :event-notification :due-now? true})]
```

Comment:
I would perhaps refactor `get-entries` to add a `get-due-entries` since only it is used in the business code, don't have to give the `:due-now?` then.

Reply:
done